mlflow/store/artifact/optimized_s3_artifact_repo.py

```diff
@@ -55,15 +57,20 @@ def __init__(
         self._region_name = self._get_region_name()
 
     def _get_region_name(self):
-        # note: s3 client enforces path addressing style for get_bucket_location
         temp_client = _get_s3_client(
             addressing_style="path",
             access_key_id=self._access_key_id,
             secret_access_key=self._secret_access_key,
             session_token=self._session_token,
             s3_endpoint_url=self._s3_endpoint_url,
         )
-        return temp_client.get_bucket_location(Bucket=self.bucket)["LocationConstraint"]
+        try:
+            return temp_client.head_bucket(Bucket=self.bucket)["BucketRegion"]
+        except ClientError as error:
+            return error.response["ResponseMetadata"]["HTTPHeaders"]["x-amz-bucket-region"]
+        except Exception as e:
```

Comment:
Could we just not catch the generic `Exception` case & just let it throw? Doesn't seem to add anything to log a warning & rethrow